kratos/containers/container_expression/expressions/unary/unary_slice_expression.cpp

```diff
@@ -0,0 +1,76 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <sstream>
+
+// Project includes
+
+// Include base h
+#include "unary_slice_expression.h"
+
+namespace Kratos {
+
+UnarySliceExpression::UnarySliceExpression(
+    Expression::Pointer pExpression,
+    const IndexType Offset,
+    const IndexType Stride)
+    : Expression(pExpression->NumberOfEntities()),
+      mpSourceExpression(pExpression),
+      mOffset(Offset),
+      mStride(Stride),
+      mSourceStride(mpSourceExpression->GetItemComponentCount())
+{
+    KRATOS_ERROR_IF(Offset + Stride > mSourceStride)
+        << "Offset + stride is greater than the maximum stride of the given expression. [ Offset = "
+        << Offset << ", stride = " << Stride << ", expression stride = "
+        << mSourceStride << " ].\n" << "Source expression:\n"
+        << *mpSourceExpression << "\n";
+}
+
+Expression::Pointer UnarySliceExpression::Create(
+    Expression::Pointer pExpression,
+    const IndexType Offset,
+    const IndexType Stride)
+{
+    return Kratos::make_intrusive<UnarySliceExpression>(std::move(pExpression), Offset, Stride);
+}
+
+double UnarySliceExpression::Evaluate(
+    const IndexType EntityIndex,
+    const IndexType EntityDataBeginIndex,
+    const IndexType ComponentIndex) const
+{
+    return mpSourceExpression->Evaluate(EntityIndex, EntityIndex * mSourceStride, ComponentIndex + mOffset);
+}
+
+const std::vector<std::size_t> UnarySliceExpression::GetItemShape() const
+{
+    // Following will always convert shape = [1] to shape = [].
+    // Hence, if slicing is used, the output for scalars always will be
+    // shape = [] irrespective of the input scalar shape. If they still want to
+    // have the shape = [1], then UnaryReshapeExpression may be used.
+    if (mStride == 1) {
+        return {};
+    } else {
+        return {mStride};
+    }
+}
```

Comment:
on second thought, why is the shape empty if the stride is 1? An empty shape would mean an empty expression.

Reply:
it is following the same as in numpy. numpy also return an empty shape if it is a scalar.